src/app/domain/limits/OauthClientsLimitService.ts

```diff
@@ -0,0 +1,113 @@
+import { EventEmitter } from 'events'
+
+import { NavigationProp, ParamListBase } from '@react-navigation/native'
+import type {
+  WebViewOpenWindowEvent,
+  WebViewNavigation
+} from 'react-native-webview/lib/WebViewTypes'
+
+import type CozyClient from 'cozy-client'
+import { deconstructCozyWebLinkWithSlug } from 'cozy-client'
+import Minilog from 'cozy-minilog'
+
+import { routes } from '/constants/routes'
+import { navigateToApp } from '/libs/functions/openApp'
+import { navigate, navigationRef } from '/libs/RootNavigation'
```

Comment:
I've thought about this recently and I've come to the conclusion a lot of issues we have with navigation come from react-decoupled calls, that don't fit well with the view lifecycles. I started to avoid as much as possible calling them from outside components in the sharing feature. What do you think?

Reply:
I tried to use the `navigation` param only, but the problem is that `navigation.getCurrentRoute()` will always return the home's route here, even if cozy-settings is on focus. Seems like the `navigation` prop's context is bind to where the component is displayed and not where the app is.

When using `navigationRef.getCurrentRoute()`, then I correctly get the cozyapp route when cozy-settings is displayed.